src/login/LoginPage.jsx

```diff
@@ -1,395 +1,318 @@
 import {
-  FormGroup, InstitutionLogistration, PasswordField, RedirectLogistration,
-  RenderInstitutionButton, SocialAuthProviders, ThirdPartyAuthAlert,
+  FormGroup,
+  InstitutionLogistration,
+  PasswordField,
+  RedirectLogistration,
+  ThirdPartyAuthAlert,
 } from '../common-components';
 import { getThirdPartyAuthContext } from '../common-components/data/actions';
 import { thirdPartyAuthContextSelector } from '../common-components/data/selectors';
 import EnterpriseSSO from '../common-components/EnterpriseSSO';
+import ThirdPartyAuth from '../common-components/ThirdPartyAuth';
 import {
-  DEFAULT_STATE, ENTERPRISE_LOGIN_URL, PENDING_STATE, RESET_PAGE,
+  DEFAULT_STATE, PENDING_STATE, RESET_PAGE,
 } from '../data/constants';
 import {
   getActivationStatus,
   getAllPossibleQueryParams,
   getTpaHint,
   getTpaProvider,
   updatePathWithQueryParams,
-  windowScrollTo,
 } from '../data/utils';
 import ResetPasswordSuccess from '../reset-password/ResetPasswordSuccess';
 
-class LoginPage extends React.Component {
-  constructor(props, context) {
-    super(props, context);
-    this.state = {
-      password: this.props.loginFormData.password,
-      emailOrUsername: this.props.loginFormData.emailOrUsername,
-      errors: {
-        emailOrUsername: this.props.loginFormData.errors.emailOrUsername,
-        password: this.props.loginFormData.errors.password,
-      },
-      isSubmitted: false,
-    };
-    this.queryParams = getAllPossibleQueryParams();
-    this.tpaHint = getTpaHint();
-  }
+const LoginPage = (props) => {
+  const {
+    backedUpFormData,
+    loginErrorCode,
+    loginErrorContext,
+    loginResult,
+    shouldBackupState,
+    thirdPartyAuthContext: {
+      providers,
+      currentProvider,
+      secondaryProviders,
+      finishAuthUrl,
+      platformName,
+      errorMessage: thirdPartyErrorMessage,
+    },
+    thirdPartyAuthApiStatus,
+    institutionLogin,
+    showResetPasswordSuccessBanner,
+    submitState,
+    // Actions
+    backupFormState,
+    handleInstitutionLogin,
+    getTPADataFromBackend,
+  } = props;
+  const { formatMessage } = useIntl();
+  const activationMsgType = getActivationStatus();
+  const queryParams = useMemo(() => getAllPossibleQueryParams(), []);
+
+  const [formFields, setFormFields] = useState({ ...backedUpFormData.formFields });
+  const [errorCode, setErrorCode] = useState({ type: '', count: 0, context: {} });
+  const [errors, setErrors] = useState({ ...backedUpFormData.errors });
+  const tpaHint = getTpaHint();
 
-  componentDidMount() {
+  useEffect(() => {
     sendPageEvent('login_and_registration', 'login');
-    const payload = { ...this.queryParams };
+  }, []);
 
-    if (this.tpaHint) {
-      payload.tpa_hint = this.tpaHint;
+  useEffect(() => {
+    const payload = { ...queryParams };
+    if (tpaHint) {
+      payload.tpa_hint = tpaHint;
     }
-    this.props.getThirdPartyAuthContext(payload);
-    this.props.loginRequestReset();
```

Comment:
i can't find it in new implementation. Dont we need this?

Reply:
Its renamed to `getTPADataFromBackend`. Please see the map props at the end of the file.

Reply:
i was asking about `loginRequestReset`

Reply:
`loginRequestReset()` has been replaced with `backupLoginForm` in Logistration. The primary objective of loginRequestReset() was to reset the error banner on the Login page upon toggling between Login and Registration. However, the current configuration of the backup function doesn't back up login errors. Consequently, when users switch from login to register and back to login, the error banner remains hidden, as it is not being backed up.